Gems/EditorPythonBindings/Code/Source/PythonMarshalTuple.cpp

```diff
@@ -0,0 +1,309 @@
+        AZStd::vector<PyObject*> items;
+
+        if (IsValidList(pyObj))
+        {
+            // Python list, just grab raw pointers to each object.
+            pybind11::list pyList(pyObj);
+            for (auto& item : pyList)
+            {
+                items.push_back(item.ptr());
+            }
+        }
+        else if (IsValidTuple(pyObj))
+        {
+            // Python tuple, just grab raw pointers to each object.
+            pybind11::tuple pyTuple(pyObj);
+            for (auto& item : pyTuple)
+            {
+                items.push_back(item.ptr());
+            }
+        }
+        else if (IsCompatibleProxy(pyObj))
+        {
+            // Python Proxy Object that's a C++ tuple. This is a bit more complicated, because there's no easy way to detect
+            // the number of properties and get them in the right order. 
+            // OnDemandReflection<AZStd::tuple<T...>> exposes "GetN" in the proxy object, so we'll keep calling that with increasing
+            // numbers until it stops working.
+            EditorPythonBindings::PythonProxyObject* proxy = pybind11::cast<EditorPythonBindings::PythonProxyObject*>(pyObj);
+            bool propertyFound = true;
+            do
+            {
+                // Generate method names like Get0(), Get1(), Get2(), etc. and call them.
+                constexpr AZStd::size_t MaxPropertyNameSize = 32;
+                auto propertyName = AZStd::fixed_string<MaxPropertyNameSize>::format("Get%zu", items.size());
+                auto item = proxy->Invoke(propertyName.c_str(), {});
+
+                // For each item that's returned, save both a copy of the object to keep it from deallocating, and the raw pointer
+                // that we'll use for the conversion step.
+                if (!item.is_none())
+                {
+                    proxyItems.push_back(item);
+                    items.push_back(item.ptr());
+                }
+                else
+                {
+                    propertyFound = false;
+                }
+            } while (propertyFound);
+        }
+
+        if (elements.size() != items.size())
+        {
+            AZ_Error("python", false, "Tuple requires %zu elements but received %zu elements.", elements.size(), items.size());
+            return AZStd::nullopt;
+        }
+
+        // For each object found, create a copy of the value as the correct element in the C++ tuple.
+        // Also, save the pointers for each allocation that we do so that we can free them in the case of an error during conversion.
+        AZStd::vector<void*> reservedElements;
+        for (size_t itemIdx = 0; itemIdx < elements.size(); itemIdx++)
+        {
+            // Allocate space for each element.
+            void* reserved = tupleDataContainer->ReserveElement(tupleInstance.m_address, elements[itemIdx]);
+            AZ_Assert(reserved, "Could not allocate tuple's element %zu via ReserveElement()", itemIdx);
```

Comment:
Do not use AZ_Assert() in Python C code since this will just look like a crash to a Python developer. It is encouraged to use AZ_Error() to bubble up a description of what the Python script needs to do to fix it.

Reply:
Fixed!